lib/honeykiq/server_middleware.rb

```diff
@@ -25,16 +26,24 @@ def extra_fields
 
     private
 
-    def call_event(event, msg, queue_name)
-      event.add(default_fields(msg, queue_name))
+    def start_span(name:)
+      if @honey_client
+        @honey_client.event
+      else
+        Honeycomb.start_span(name: name)
+      end
```

Comment:
Should we mention this change in the README? The fact that `honey_client` is now optional to pass in on initialisation, and that it will fall back to the globally configured `Honeycomb` client?

Should we move `honeycomb-beeline` to be a runtime dependency? Or add some safety check in case `Honeycomb` is not defined?

Reply:
Not adding `honeycomb-beeline` as a dependency so that we can use it without installing the beeline since the beeline does a bunch of auto monkey-patching which could break things.